File: clients/python/examples/list_models.py
#!/usr/bin/env python3
"""
List all available models using the modern sigo_client.
"""

from sigo_client import SigoClient


def main():
    client = SigoClient("http://127.0.0.1:9080")

    if not client.ping():
        print("❌ sigoREST server is not responding on http://127.0.0.1:9080")
        print("   Start it with: ./sigoREST/sigoREST -q")
        return

    print("📋 sigoREST Models (v2 client)\n")
    print(f"✅ Connected | Server Status: OK\n")

    health = client.health()
    print(f"Available Models : {health.available_models}")
    print(f"Memory configured: {health.memory_set}")
    print("-" * 60)

    models = client.list_models()

    # Group by family
    from collections import defaultdict
    groups = defaultdict(list)

    for model in models:
        if "claude" in model.id.lower() or model.shortcode.startswith("cl"):
            groups["Claude"].append(model)
        elif any(x in model.id.lower() for x in ["kimi", "moonshot"]):
            groups["Moonshot / Kimi"].append(model)
        elif "ollama" in model.id.lower():
            groups["Ollama (Local)"].append(model)
        else:
            groups["Other"].append(model)

    for group_name, model_list in sorted(groups.items()):
        print(f"\n🔹 {group_name} ({len(model_list)} models)")
        print(f"{'Shortcode':<12} {'ID':<35} {'Input':>6} {'Output':>6}")
        print("-" * 65)
        for m in sorted(model_list, key=lambda x: x.shortcode or x.id):
            short = m.shortcode or m.id.split('/')[-1][-8:]
            print(f"{short:<12} {m.id:<35} ${m.input_cost:>5.2f} ${m.output_cost:>5.2f}")
        print()


if __name__ == "__main__":
    main()

File: clients/javascript/examples/listmodels.js
#!/usr/bin/env node
/**
 * List available models example
 *
 * This example shows how to get information about available models.
 */

import { SigoClient } from '../client.js';

function getProvider(model) {
  if (model.endpoint.includes('mammouth')) return 'Mammoth.ai';
  if (model.endpoint.includes('moonshot')) return 'Moonshot';
  if (model.endpoint.includes('z.ai')) return 'Z.ai';
  if (model.shortcode.startsWith('ollama-')) return 'Ollama (Local)';
  return 'Other';
}

async function main() {
  const client = new SigoClient('http://127.0.0.1:9080');

  const isAlive = await client.ping();
  if (!isAlive) {
    console.error('❌ sigoREST server is not responding');
    process.exit(1);
  }

  console.log('📋 Available Models\n');

  // Get health info
  const health = await client.health();
  console.log(`Server Status: ${health.status}`);
  console.log(`Available Models: ${health.available_models}`);
  console.log(`Memory Set: ${health.memory_set}`);
  console.log();

  // List all models
  const models = await client.listModels();

  // Group by provider
  const providers = {};
  for (const model of models) {
    const provider = getProvider(model);
    if (!providers[provider]) {
      providers[provider] = [];
    }
    providers[provider].push(model);
  }

  // Sort providers and display
  const sortedProviders = Object.keys(providers).sort();
  for (const provider of sortedProviders) {
    console.log(`\n--- ${provider} ---`);
    console.log(`${'ID'.padEnd(30)} ${'Shortcode'.padEnd(15)} ${'Input'.padStart(8)} ${'Output'.padStart(8)}`);
    console.log('-'.repeat(65));

    // Sort models by shortcode
    const sortedModels = providers[provider].sort((a, b) =>
      a.shortcode.localeCompare(b.shortcode)
    );

    for (const m of sortedModels) {
      console.log(
        `${m.id.padEnd(30)} ${m.shortcode.padEnd(15)} ` +
        `$${m.input_cost.toFixed(2).padStart(6)} $${m.output_cost.toFixed(2).padStart(6)}`
      );
    }
  }
}

main().catch(err => {
  console.error('❌ Error:', err.message);
  process.exit(1);
});
